ttnn/cpp/ttnn/deprecated/tt_lib/csrc/tt_lib_bindings_tensor_impl.hpp

```diff
@@ -82,31 +84,79 @@ void bind_op_with_mem_config_and_dtype(py::module_ &module, std::string op_name,
             "{0}", "Output tensor data type", "DataType", "Default is None (Use input dtype)", "No")doc",
             dtype_name
         );
-        module.def(op_name.c_str(), f,
-            std::forward<Extra>(extra)..., py::arg(mem_config_name.c_str()).noconvert() = operation::DEFAULT_OUTPUT_MEMORY_CONFIG, py::arg(dtype_name.c_str()).noconvert() = std::nullopt, docstring.c_str()
-        );
+        if constexpr (kernel_config) {
+            module.def(
+                op_name.c_str(),
+                f,
+                std::forward<Extra>(extra)...,
+                py::arg(mem_config_name.c_str()).noconvert() = operation::DEFAULT_OUTPUT_MEMORY_CONFIG,
+                py::arg(dtype_name.c_str()).noconvert() = std::nullopt,
+                py::arg(kernel_config_name.c_str()).noconvert() = std::nullopt,
+                docstring.c_str());
+        } else {
+            module.def(
+                op_name.c_str(),
+                f,
+                std::forward<Extra>(extra)...,
+                py::arg(mem_config_name.c_str()).noconvert() = operation::DEFAULT_OUTPUT_MEMORY_CONFIG,
+                py::arg(dtype_name.c_str()).noconvert() = std::nullopt,
+                docstring.c_str());
+        }
```

Comment:
This looks unclean to me since the function is named explicitly `bind_op_with_mem_config_and_dtype` and now we are adding more args to it and the code is forked. Maybe this is a byproduct of bad code to begin with, so I don't have a strong opinion on this.

Reply:
I don't like it as well, I assumed that this piece of code is going to history with move to ttnn?

Reply:
Yes, this will be deleted at the end of the fragment so it doesn't need to be clean so long as the ttnn version is clean.

Reply:
Ok sounds good.